changes/tasks.py

```diff
@@ -542,6 +542,26 @@ def switch_channel(self, change):
                     lang=sub["lang"],
                     schedule=sub["schedule"],
                 )
+                reg = (
+                    Registration.objects.filter(registrant_id=change.registrant_id)
+                    .order_by("-created_at")
+                    .first()
+                )
+                weeks = utils.get_pregnancy_week(utils.get_today(), reg.data["edd"])
+                msgset_short_name = utils.get_messageset_short_name(
+                    "whatsapp_service_info", reg.source.authority, weeks
+                )
+                r = utils.get_messageset_schedule_sequence(msgset_short_name, weeks)
+                msgset_id, msgset_schedule, next_sequence_number = r
+
+                SubscriptionRequest.objects.create(
+                    identity=reg.registrant_id,
+                    messageset=msgset_id,
+                    next_sequence_number=next_sequence_number,
+                    lang=sub["lang"],
+                    schedule=msgset_schedule,
+                )
```

Comment:
This seems to run for every active SMS subscription, which means if the mother has both a `momconnect_prebirth` and `pmtct_prebirth` subscriptions, then she'll get two subscriptions to the service info message set.

We should probably only create the service info subscription for each momconnect pre/post birth message set

Reply:
Good catch, thanks!